base/array.jl

```diff
@@ -304,6 +304,7 @@ copyto!(dest::Memory, doffs::Integer, src::Array, soffs::Integer, n::Integer) =
 copyto!(dest::Array{T}, doffs::Integer, src::Array{T}, soffs::Integer, n::Integer) where {T} = _copyto_impl!(dest, doffs, src, soffs, n)
 
 function _copyto_impl!(dest::Union{Array,Memory}, doffs::Integer, src::Union{Array,Memory}, soffs::Integer, n::Integer)
+    @_propagate_inbounds_meta
```

Comment:
Is the inbounds being propagated to `unsafe_copyto!`, or is this only to disable the `@boundscheck`s? In the latter case, perhaps this should only be an `@inline`?